mongodbatlas_test.go

```diff
@@ -194,6 +200,51 @@ func TestAcceptanceDatabaseUser_CreateUser(t *testing.T) {
 	assertCredsExists(t, projectID, publicKey, privateKey, createResp.Username, password, connURL, testMongoDBAtlasRole)
 }
 
+func TestAcceptanceDatabaseUser_CreateX509User(t *testing.T) {
+	if !runAcceptanceTests {
+		t.SkipNow()
+	}
+
+	publicKey := os.Getenv(envVarAtlasPublicKey)
+	privateKey := os.Getenv(envVarAtlasPrivateKey)
+	projectID := os.Getenv(envVarAtlasProjectID)
+	connURL := os.Getenv(envVarAtlasClusterName)
+
+	connectionDetails := map[string]interface{}{
+		"public_key":  publicKey,
+		"private_key": privateKey,
+		"project_id":  projectID,
+	}
+
+	db := new()
+	defer dbtesting.AssertClose(t, db)
+
+	initReq := dbplugin.InitializeRequest{
+		Config: connectionDetails,
+	}
+
+	dbtesting.AssertInitialize(t, db, initReq)
+
+	subject := "CN=token_my-test-role_1684350512"
+	createReq := dbplugin.NewUserRequest{
+		CredentialType: dbplugin.CredentialTypeClientCertificate,
+		UsernameConfig: dbplugin.UsernameMetadata{
+			DisplayName: "testcreate",
+			RoleName:    "test",
+		},
+		Statements: dbplugin.Statements{
+			Commands: []string{testMongoDBAtlasX509Role},
+		},
+		Subject:    subject,
+		Expiration: time.Now().Add(time.Minute),
+	}
+
+	createResp := dbtesting.AssertNewUser(t, db, createReq)
+	defer deleteAtlasDBUser(t, projectID, publicKey, privateKey, createResp.Username)
```

Comment:
nit: consider `t.Cleanup` instead of `defer` in case of panics

Reply:
To stay consistent, I think I would rather leave it like this. Unless you think it would be more helpful to use the cleanup?

Reply:
nope! agree that consistency is more important